### PR TITLE
DROOLS-3262 DROOLS-3281 [DMN Designer] Decision Table In/Out clause fixs

### PR DESCRIPTION
**Pull request by tarilabs**
DROOLS-3281
[DMN Designer] The output name is valorized for single output decision
table

DROOLS-3262
[DMN Designer] The `defaultOutputValue` and the `outputValue` are being
persisted when the value is empty in the UI

/cc @jomarko @karreiro @manstis

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/InputClausePropertyConverter.java [hunk 1]
```diff
@@ -57,10 +57,11 @@ public static InputClause wbFromDMN(final org.kie.dmn.model.api.InputClause dmn)
             expression.setParent(result);
         }
         result.setInputExpression(expression);
-        if (inputValues != null) {
+
+        if (inputValues != null && !inputValues.getText().isEmpty()) {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/InputClausePropertyConverter.java]: You could use `org.kie.workbench.common.stunner.core.util.StringUtils.nonEmpty(..)` (but how was you to know!?!)

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/OutputClausePropertyConverter.java [hunk 2]
```diff
@@ -57,16 +57,19 @@ public static OutputClause wbFromDMN(final org.kie.dmn.model.api.OutputClause dm
         result.setId(wb.getId().getValue());
         result.setName(wb.getName());
         result.setDescription(DescriptionPropertyConverter.dmnFromWB(wb.getDescription()));
+
         UnaryTests outputValues = UnaryTestsPropertyConverter.dmnFromWB(wb.getOutputValues());
-        if (outputValues != null) {
+        if (outputValues != null && !outputValues.getText().isEmpty()) {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/OutputClausePropertyConverter.java]: You could use `org.kie.workbench.common.stunner.core.util.StringUtils.nonEmpty(..)` (but how was you to know!?!)

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/OutputClausePropertyConverter.java [hunk 3]
```diff
@@ -57,16 +57,19 @@ public static OutputClause wbFromDMN(final org.kie.dmn.model.api.OutputClause dm
...
             outputValues.setParent(result);
+            result.setOutputValues(outputValues);
         }
-        result.setOutputValues(outputValues);
+
         LiteralExpression defaultOutputEntry = LiteralExpressionPropertyConverter.dmnFromWB(wb.getDefaultOutputEntry());
-        if (defaultOutputEntry != null) {
+        if (defaultOutputEntry != null && !defaultOutputEntry.getText().isEmpty()) {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/OutputClausePropertyConverter.java]: You could use `org.kie.workbench.common.stunner.core.util.StringUtils.nonEmpty(..)` (but how was you to know!?!)

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DecisionTablePropertyConverter.java [hunk 4]
```diff
@@ -95,6 +95,9 @@ public static DecisionTable wbFromDMN(final org.kie.dmn.model.api.DecisionTable
             }
             result.getOutput().add(c);
         }
+        if (result.getOutput().size() == 1) {
+            result.getOutput().get(0).setName(null); // DROOLS-3281
```
Comment (karreiro) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/definition/v1_1/DecisionTablePropertyConverter.java]: I see comments with JIRAs numbers as a temporary thing in the code base. I would replace this line with something like:
```suggestion
            // The output name is valorized for single output decision table
            result.getOutput().get(0).setName(null);
```
I think the message above gives more context to the reader (and avoid an indirection to the JIRA).

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java [hunk 5]
```diff
@@ -319,6 +323,31 @@ public void testLoan() throws IOException {
...
+        LOG.debug("{}", dmnResult);
+        assertThat(dmnResult.hasErrors(), is(false));
+        assertThat(dmnResult.getDecisionResultByName("my decision").getEvaluationStatus(), is(DecisionEvaluationStatus.SUCCEEDED));
+        assertThat(dmnResult.getDecisionResultByName("my decision").getResult(), is("negative"));
+
+        org.kie.dmn.model.api.DecisionTable dmnDT = (DecisionTable) model.getDecisionByName("my decision").getDecision().getExpression();
+        assertThat(dmnDT.getInput().get(0).getInputValues(), nullValue()); // DROOLS-3262
+        assertThat(dmnDT.getOutput().get(0).getOutputValues(), nullValue()); // DROOLS-3262
+        assertThat(dmnDT.getOutput().get(0).getDefaultOutputEntry(), nullValue()); // DROOLS-3262
+        assertThat(dmnDT.getOutput().get(0).getName(), nullValue()); // DROOLS-3281
```
Comment (karreiro) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/test/java/org/kie/workbench/common/dmn/backend/DMNMarshallerTest.java]: ..the same thing for the JIRA numbers above.

**Comment by tarilabs** [comment]
> I reported the typeRef inconsistency https://issues.jboss.org/browse/DROOLS-3289,

:+1:

**Pull request merged**